src/comments.js

```diff
@@ -504,19 +528,33 @@ function handleImportSpecifierComments(enclosingNode, comment) {
 }
 
 function handleCallExpressionComments(precedingNode, enclosingNode, comment) {
-  if (enclosingNode && enclosingNode.type === "CallExpression" &&
-      precedingNode && enclosingNode.callee === precedingNode &&
-      enclosingNode.arguments.length > 0) {
+  if (
+    enclosingNode &&
+    enclosingNode.type === "CallExpression" &&
+    precedingNode &&
+    enclosingNode.callee === precedingNode &&
+    enclosingNode.arguments.length > 0
+  ) {
     addLeadingComment(enclosingNode.arguments[0], comment);
     return true;
   }
   return false;
 }
 
-function handleUnionTypeComments(precedingNode, enclosingNode, followingNode, comment) {
-  if (enclosingNode && enclosingNode.type === 'UnionTypeAnnotation' &&
-      precedingNode && precedingNode.type === 'ObjectTypeAnnotation' &&
-      followingNode && followingNode.type === 'ObjectTypeAnnotation') {
+function handleUnionTypeComments(
+  precedingNode,
+  enclosingNode,
+  followingNode,
+  comment
+) {
+  if (
+    enclosingNode &&
+    enclosingNode.type === "UnionTypeAnnotation" &&
+    precedingNode &&
+    precedingNode.type === "ObjectTypeAnnotation" &&
+    followingNode &&
+    followingNode.type === "ObjectTypeAnnotation"
+  ) {
```

Comment:
It's interesting, if I put parenthesis around to group things, it works except for the first one.

```js
function f() {
  if (
    (precedingNode &&
    precedingNode.type === "FunctionTypeParam") &&
    (enclosingNode &&
    enclosingNode.type === "FunctionTypeAnnotation") &&
    (followingNode &&
    followingNode.type !== "FunctionTypeParam")
  ) {
    addTrailingComment(precedingNode, comment);
    return true;
  }
}
```

outputs

```js
function f() {
  if (
    precedingNode &&
    precedingNode.type === "FunctionTypeParam" &&
    (enclosingNode && enclosingNode.type === "FunctionTypeAnnotation") &&
    (followingNode && followingNode.type !== "FunctionTypeParam")
  ) {
    addTrailingComment(precedingNode, comment);
    return true;
  }
}
```